v2/audiences_test.go

```diff
@@ -0,0 +1,190 @@
+package courier_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/trycourier/courier-go/v2"
+)
+
+func TestPutAudience(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(
+		func(rw http.ResponseWriter, req *http.Request) {
+			assert.Equal(t, "/audiences/software-engineers-from-sf", req.URL.String())
+			rw.Header().Add("Content-Type", "application/json")
+			rsp := `
+			{
+				"audience": {
+					"id": "software-engineers-from-sf",
+					"name": "Software Engineers From SF",
+					"description": "Updated descriptionss",
+					"created_at": "2022-03-22T19:07:32.066Z",
+					"updated_at": "2022-03-24T00:32:11.122Z",
+					"filter": {
+						"path": "title",
+						"value": "Software Engineer",
+						"operator": "EQ"
+					}
+				}
+			}`
+			_, _ = rw.Write([]byte(rsp))
+
+		}))
+	defer server.Close()
+
+	t.Run("Put Audience", func(t *testing.T) {
```

Comment:
nit: Please use a more descriptive title for the test name. Example: Put Audience Happy Path and then I'd like to see one for the failed path as well.